Atos20/Crate
# Nd visual selection

What’s this PR do?  
- Allows a user to select multiple products in each category with visual feedback.
- Allows a user to un-select multiple products in each category with visual feedback.
- Set up conditional logic for a user to not be able to finish the survey with no answers selected.
- Adds conditional logic for a user to see feedback from graphics and changing text to know when a product has been selected/un-selected.
- Changes the click event from being on the ImageTile to the Button.
- Changes StyleSurvey into a HOC withRouter and connect to redirect a user later while using connect to access showing feedback to the user.
 
Where should the reviewer start?  
- Git pull this branch.
- `npm start` inside code directory to start backend and frontend local servers at the same time.
 
How should this be manually tested?  
- Log in with a user after navigating to localhost:3000
- After logging in, go to the Crate route and click on the `+subscribe` button.
- Test going through the survey without selecting any answers and clicking submit for a message to not be able to complete without selecting something.
- Test selecting and un-selecting products.
- Select with products selected to see the current break down of the categories.
 
Any background context you want to provide?  
- n/a
 
What are the relevant tickets?  
- fix #13
- fix #39
- fix #40
- fix #42

Screenshots (if appropriate)  
- n/a
 
Questions: 
- n/a
